### ISSUE TITLE
loss.backward() cause malloc: *** error for object 0x7fb849370210: incorrect checksum for freed object

### ISSUE THREAD

**Issue by bjmajic** [pytorch/pytorch#16646]
I build a net: resnet18 + ctc loss, when run loss.backward(), there will be a error:python(17267,0x700007df0000) malloc: *** error for object 0x7fb849370210: incorrect checksum for freed object
i not find any method to resolve it?

note: my training date size is not equal, this mean :the widht of image is variable, and the height of imgage is 32,    the label length is variable

========= my net.py is ===================
# coding=utf-8
import torch
import torch.nn as nn
import torch.nn.functional as F

# 直接定义双层LSTM
class BiLSTM(nn.Module):
    def __init__(self, num_in, num_hidden, num_out):
        super(BiLSTM, self).__init__()
        self.LSTM = nn.LSTM(input_size=num_in, hidden_size=num_hidden, num_layers=2,
                            bidirectional=True, batch_first=False)
        self.embeding = nn.Linear(num_hidden * 2, num_out)

    def forward(self, input):
        n, c, h, w = input.size()
        input = input.permute(0, 3, 2, 1)
        input = input.reshape(n, w, h*c)
        input = input.permute(1, 0, 2)
        recurrent, _ = self.LSTM(input)

        out_put = self.embeding(recurrent)
        return out_put

class ResidualBlock(nn.Module):
    def __init__(self, inchannel, outchannel, stride=(1, 1)):
        super(ResidualBlock, self).__init__()
        self.left = nn.Sequential(
            nn.Conv2d(inchannel, outchannel, kernel_size=3, stride=stride, padding=1, bias=False),
            nn.BatchNorm2d(outchannel),
            nn.ReLU(inplace=True),
            nn.Conv2d(outchannel, outchannel, kernel_size=3, stride=1, padding=1, bias=False),
            nn.BatchNorm2d(outchannel)
        )
        self.shortcut = nn.Sequential()
        if stride != (1, 1) or inchannel != outchannel:
            self.shortcut = nn.Sequential(
                nn.Conv2d(inchannel, outchannel, kernel_size=1, stride=stride, bias=False),
                nn.BatchNorm2d(outchannel)
            )

    def forward(self, x):
        out = self.left(x)
        out += self.shortcut(x)
        out = F.relu(out)
        return out

class ResNet(nn.Module):
    def __init__(self, ResidualBlock, num_classes=10, test=False):
        super(ResNet, self).__init__()
        self.inchannel = 64
        self.conv1 = nn.Sequential(
            nn.Conv2d(3, 64, kernel_size=3, stride=1, padding=1, bias=False),
            nn.BatchNorm2d(64),
            nn.ReLU()
        )
        self.layer1 = self.make_layer(ResidualBlock, 64, 2, stride=(1, 1))
        self.layer2 = self.make_layer(ResidualBlock, 128, 2, stride=(2, 2))
        self.layer3 = self.make_layer(ResidualBlock, 256, 2, stride=(2, 2))
        self.layer4 = self.make_layer(ResidualBlock, 512, 2, stride=(2, 1))

        # self.rnn = nn.Sequential(
        #     BidirectionalLSTM(512, 512, 512),
        #     BidirectionalLSTM(512, 512, num_classes),)
        self.rnn = BiLSTM(512, 512, num_classes)
        # self.fc = nn.Linear(512, num_classes)
        self.test = test

    def make_layer(self, block, channels, num_blocks, stride):
        strides = [stride] + [(1, 1)]*(num_blocks - 1)
        layers = []
        for stride in strides:
            layers.append(block(self.inchannel, channels, stride))
            self.inchannel = channels
        return nn.Sequential(*layers)

    def forward(self, x):
        out = self.conv1(x)
        out = self.layer1(out)
        out = self.layer2(out)
        out = self.layer3(out)
        out = self.layer4(out)
        out = F.avg_pool2d(out, (4, 1))

        if self.test:
            return out

        # out = out.view(out.size(0), -1)
        # out = self.fc(out)
        out = self.rnn(out)
        return out

def resnet18():
    return ResNet(ResidualBlock)

========== my train code is ==========
    def train_batch(self, epoch):
        self.model.train()
        for batch_idx, (img, label, width, lens) in enumerate(self.train_loader):
            self.optimizer.zero_grad()
            preds = self.model(img)
            log_probs = preds.log_softmax(2).detach().requires_grad_()
            loss = self.criterion(log_probs, label, width, lens)
            print(loss)
            loss.backward()
            self.optimizer.step()  # 更新参数
            if (batch_idx+1) % self.log_interval == 0:
                lr = self.scheduler.get_lr()[0]
                src_str = self.label_convert.get_str_from_label(lables=label)
                preds_size = torch.IntTensor([preds.size(0)] * preds.size(1))
                _, preds = preds.max(2)
                preds = preds.squeeze(2)
                preds = preds.transpose(1, 0)
                sim_preds = self.label_convert.decode(preds, preds_size, raw=False)
                n_correct = 0
                for pred, target in zip(sim_preds, src_str):
                    if pred == target:
                        n_correct += 1
                distance_value = util.edit_distance(src_str, sim_preds)
                localtime = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
                print('[%s]\tTrain Epoch: {%d/%d}\tLoss:%.6f, acc: %.2f, error: %.2f, lr = %.4f'
                      % (str(localtime), epoch, batch_idx, loss, n_correct, distance_value, lr))
            if (batch_idx+1) % self.save_interval == 0:
                new_save_name = 'resnet18_%d_%d.pth' % (epoch, batch_idx+1)
                torch.save(self.state, new_save_name)
                util.update_file(self.save_path, new_save_name)
        new_save_name = 'resnet18_%d.pth' % epoch
        torch.save(self.state, new_save_name)
        util.update_file(self.save_path, new_save_name)

**Comment by bjmajic** [comment]
my os is mocOS

**Comment by zou3519** [comment]
@bjmajic where are you calling CTCLoss?

**Comment by bjmajic** [comment]
@zou3519 pytorch 1.0 has the ctcloss, and i call it in 
preds = self.model(img)
log_probs = preds.log_softmax(2).detach().requires_grad_()
loss = self.criterion(log_probs, label, width, lens)
print(loss)
loss.backward()

**Comment by t-vi** [comment]
Can you output the `log_probs`, `label` (at least the shapes and devices for those), and the `width` and `lens`, please? Thank you!

**Comment by bjmajic** [comment]
@t-vi thank you.   It was my fault， not a bug.  it caused by dismatch of labels dim and features dim.

**Comment by apaszke** [comment]
Even if it was a mismatch we should have an error message and never corrupt the memory

**Comment by t-vi** [comment]
So we (@ashermancinelli) improved the checks for input/target lengths recently and it might be worth trying if that helps. We do not check that the labels fall into the valid range (neither, I believe, do the scatter ops).
